internal/controller/ppnd/ppnd_pause.go

```diff
@@ -103,28 +103,30 @@ func (pm *PauseModule) PausePipeline(ctx context.Context, c client.Client, pipel
 // resume pipeline
 // lock the maps while we change pipeline lifecycle so nobody changes their pause request
 // while we run; otherwise, they may think they are pausing the pipeline while it's running
-func (pm *PauseModule) RunPipelineIfSafe(ctx context.Context, c client.Client, pipeline *unstructured.Unstructured, isbsvcName string) (bool, error) {
+func (pm *PauseModule) RunPipeline(ctx context.Context, c client.Client, pipeline *unstructured.Unstructured, isbsvcName string, force bool) error {
 	pm.lock.RLock()
 	defer pm.lock.RUnlock()
 
-	// verify that all requests are still to pause, if not we can't run right now
-	controllerPauseRequest := pm.PauseRequests[pm.GetNumaflowControllerKey(pipeline.GetNamespace())]
-	var existingPipelineSpec numaflowtypes.PipelineSpec
-	if err := util.StructToStruct(pipeline.Object["spec"], &existingPipelineSpec); err != nil {
-		return false, err
+	if !force {
+		// verify that all requests are still to pause, if not we can't run right now
+		controllerPauseRequest := pm.PauseRequests[pm.GetNumaflowControllerKey(pipeline.GetNamespace())]
+		var existingPipelineSpec numaflowtypes.PipelineSpec
+		if err := util.StructToStruct(pipeline.Object["spec"], &existingPipelineSpec); err != nil {
+			return err
+		}
+		if controllerPauseRequest != nil && *controllerPauseRequest {
+			// controller is requesting to pause - can't run
+			return fmt.Errorf("new pause request from NumaflowController, can't resume pipeline at this time; will try again later")
+		}
+
+		isbsvcPauseRequest := pm.PauseRequests[pm.GetISBServiceKey(pipeline.GetNamespace(), isbsvcName)]
+		if isbsvcPauseRequest != nil && *isbsvcPauseRequest {
+			// isbsvc is requesting to pause - can't run
+			return fmt.Errorf("new pause request from InterstepBufferService, can't resume pipeline at this time; will try again later")
```

Comment:
returning an error instead of a boolean; this will enable us to stay in the PPND "in progress strategy" so we can come back here and try again